starknet_devnet/starknet_wrapper.py

```diff
@@ -329,7 +329,7 @@ async def update_pending_state(  # pylint: disable=too-many-arguments
         )
 
         return BlockStateUpdate(
-            block_hash=None,
+            block_hash=0,
```

Comment:
I'm not sure about this line.

Reply:
```
@marshmallow_dataclass.dataclass(frozen=True)
class BlockStateUpdate(ValidatedResponseObject):
    """
    Represents a response block state update.
    """

    block_hash: Optional[int] = field(metadata=fields.optional_block_hash_metadata)
    new_root: Optional[bytes] = field(metadata=fields.optional_state_root_metadata)
    old_root: bytes = field(metadata=fields.state_root_metadata)
    state_diff: StateDiff

    def __post_init__(self):
        super().__post_init__()
        assert (self.block_hash is None) == (
            self.new_root is None
        ), "new_root must appear in state update for any block other than pending block."
```

Reply:
so we have this assert `assert (self.block_hash is None) == (self.new_root is None)`

Reply:
So block_hash and new_root need to be of the same `None`ness. Is it legal if all three values are `None` (block hash, old root, new root)? If not, maybe then either add a comment explaining that this is a dummy value, or create a new constant (similar to DUMMY_ROOT)

Reply:
Validation is not passing if `new_root` or `block_hash` is not set (strange because it's optional). And both can't be `None` so I just will add `DUMMY_PENDING_BLOCK_HASH`.